src/test/scala/org/broadinstitute/dsde/workbench/leonardo/auth/SamAuthProviderSpec.scala

```diff
@@ -0,0 +1,125 @@
+  val iamDAO = new MockGoogleIamDAO
+
+  val leo = new LeonardoService(dataprocConfig, clusterFilesConfig, clusterResourcesConfig, proxyConfig, swaggerConfig, gdDAO, iamDAO, DbSingleton.ref, system.actorOf(NoopActor.props), samAuthProvider, serviceAccountProvider, whitelist)
+
+  "should add and delete a notebook-cluster resource with correct actions for the user when a cluster is created and then destroyed" in isolatedDbTest {
+    samAuthProvider.samClient.billingProjects += (project, userInfo.userEmail) -> Set("launch_notebook_cluster")
+    // check the sam auth provider has no notebook-cluster resource
+    samAuthProvider.samClient.notebookClusters shouldBe empty
+    samAuthProvider.samClient.hasActionOnNotebookClusterResource(userInfo.userEmail, project, name1, "status") shouldBe false
+    samAuthProvider.samClient.hasActionOnNotebookClusterResource(userInfo.userEmail, project, name1, "connect") shouldBe false
+    samAuthProvider.samClient.hasActionOnNotebookClusterResource(userInfo.userEmail, project, name1, "sync") shouldBe false
+    samAuthProvider.samClient.hasActionOnNotebookClusterResource(userInfo.userEmail, project, name1, "delete") shouldBe false
+
+    // create a cluster
+    val cluster = leo.createCluster(userInfo,project,name1, testClusterRequest).futureValue
+    val clusterStatus = leo.getActiveClusterDetails(userInfo, project, name1).futureValue
+    cluster shouldEqual clusterStatus
+
+    // check the resource exists for the user and actions
+    samAuthProvider.samClient.hasActionOnNotebookClusterResource(userInfo.userEmail, project, name1, "status") shouldBe true
+    samAuthProvider.samClient.hasActionOnNotebookClusterResource(userInfo.userEmail, project, name1, "connect") shouldBe true
+    samAuthProvider.samClient.hasActionOnNotebookClusterResource(userInfo.userEmail, project, name1, "sync") shouldBe true
+    samAuthProvider.samClient.hasActionOnNotebookClusterResource(userInfo.userEmail, project, name1, "delete") shouldBe true
+
+    leo.deleteCluster(userInfo, project, name1).futureValue
+
+    samAuthProvider.samClient.notebookClusters shouldBe empty
+    samAuthProvider.samClient.hasActionOnNotebookClusterResource(userInfo.userEmail, project, name1, "status") shouldBe false
+    samAuthProvider.samClient.hasActionOnNotebookClusterResource(userInfo.userEmail, project, name1, "connect") shouldBe false
+    samAuthProvider.samClient.hasActionOnNotebookClusterResource(userInfo.userEmail, project, name1, "sync") shouldBe false
+    samAuthProvider.samClient.hasActionOnNotebookClusterResource(userInfo.userEmail, project, name1, "delete") shouldBe false
+
+    samAuthProvider.samClient.billingProjects.remove((project, userInfo.userEmail))
+
+  }
+
+  "hasProjectPermission should return true if user has project permissions and false if they do not" in isolatedDbTest {
+    samAuthProvider.samClient.billingProjects += (project, userInfo.userEmail) -> Set("launch_notebook_cluster")
```

Comment:
you do something to `samAuthProvider.samClient` at the top of each test but i'm not seeing them being cleaned up anywhere -- are you sure it's all unique data and the tests aren't interfering with each other?

I suggest defining a function that creates a new `SamAuthProvider` (and populates it, if that makes sense), and calling that at the top of each test. Keep 'em isolated.